src/api/routes/v1/nodes.ts

```diff
@@ -191,19 +191,26 @@ export async function handleNode(req: Request, res: Response): Promise<void> {
 /**
  * Handles Nodes request.
  *
- * @param _u - Unused express request.
+ * @param req - Unused express request.
  * @param res - Express response.
  */
-export async function handleNodes(_u: Request, res: Response): Promise<void> {
+export async function handleNodes(req: Request, res: Response): Promise<void> {
   try {
     if (Date.now() - cache.time > 60 * 1000) {
       await cacheNodes()
     }
 
+    const { network } = req.params
+    const nodes =
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- Necessary here
+      network == null
+        ? cache.nodes
+        : cache.nodes.filter((node) => node.networks?.includes(network))
```

Comment:
Can a single node be following both testnet and mainnet? (What would be the case where the node has multiple networks?)

Reply:
That's not possible - a node has to be specific to a network (that's what #27 fixes).

Reply:
If that's not possible, what does `node.networks` represent? (Why is it an array of network names instead of a single one?)

Reply:
It's not an array, it's a string - it's just searching the string. (I'm not quite confident enough in my fix to assume that it's rock solid)

Reply:
Ah that makes sense :)